removed 'from torch.autograd.gradcheck import zero_gradients' from th…

pstarconsult opened the pull request:
…e first cell in this NB. This was removed in prior version of Torch and causes an import reference error. After removing this import, it runs fine.

Fixes # (issue)

## Type of change

Please check all relevant options.

- [ ] Improvement (non-breaking)
- [ x] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Test Configuration**:
- OS:Ubuntu 22.02
- Python version: 3.10
- ART version or commit number - Latest Nov 2022

# Checklist

- [ x] My code follows the style guidelines of this project
- [ x] I have performed a self-review of my own code
- [x ] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [ x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ x] New and existing unit tests pass locally with my changes

beat-buesser reviewed (changes requested):
Hi @pstarconsult Thank you very much for your pull request!

It looks like your commits have not been signed-off. To add your Signed-off-by to every commit in your branch:

- Ensure you have a local copy of your branch by checking out the pull request locally via command line
- In your local branch, run: `git rebase HEAD~1 --signoff`
- Force push your changes to overwrite the branch: `git push --force-with-lease origin main`


Would it be possible to remove the warnings in the output of cells 22 and 23?

pstarconsult commented:
@beat-buesser I entered a comment about this in #1911 but thought I would mention here too. I am not sure what if anything I should do with the "11 failing, 21 successful, 1 action required, 13 expected, and 1 neutral checks" above. 

Also, it looks like I am unable to post in the Slack channel. Ideas on how we might communicate there about making contributions? Thank you!

pstarconsult commented:
Before I close out with a 'git push --force-with-lease origin main' do you want me to fix cell 22 and 23 warnings?

beat-buesser commented:
@pstarconsult The codecov are because of a bandwith issue between Codecov and GitHub, I need to restart the failing tests for now. You can either do the force push described above and add a second commit, or replace the commit with a new commit that includes the requested changes.

pstarconsult commented:
@beat-buesser I successfully ran the git commands you suggested so you should be all set there. Let me know if not. I am going to pick up more conversation with you in the slack #random channel. Thank you!

pstarconsult commented:
This is a further explanation of the commit message I just made. 

base.py at line 450 requires X values for training data. If not, this error was being generated:

"...site-packages/sklearn/base.py:450: UserWarning: X has feature names, but LogisticRegression was fitted without feature names"

The solution is found here: https://stackoverflow.com/questions/69326639/sklearn-warning-valid-feature-names-in-version-1-0

Basically, base.py is checking to make sure there is a proper "shape" of the X and Y values - if the shape is not correct, i.e. it expects a list with two values, one for X and one for Y, it throws the above warning. The code in this notebook is not forwarding this value because it is indexing the wrong row. Here is an explanation copied from the Stackoverflow link above that explains it:

"model_reg.fit(scaled_x_train, y_train[vp].values)
data_pred = model_reg.predict(scaled_x_test.values)

This first code had the warning, because scaled_x_train is a DataFrame with feature names, while scaled_x_test.values is only values, without feature names. Then, I changed to this:

model_reg.fit(scaled_x_train.values, y_train[vp].values)
data_pred = model_reg.predict(scaled_x_test.values)"

So, I changed every occurrence in this notebook here where 'fit' was being called to use the '.values' variant, for example:

'fit(X_train_iris.values, y_train_iris)' rather than 'fit(X_train_iris, y_train_iris)'

Now the function pulls in the right values, the 'shape' has two values (one now for X) and the warning above is gone.

beat-buesser commented on .gitignore:
```diff
@@ -10,6 +10,8 @@ __pycache__/
 .Python
 env/
 build/
+art-aml/
+bin/
```
```suggestion
```

beat-buesser commented on .gitignore:
```diff
@@ -111,3 +113,4 @@ demo/pics/*
 !notebooks/adaptive_defence_evaluations/*.ipynb
 !notebooks/adversarial_patch/*.ipynb
 !notebooks/art_evaluations/*.ipynb
+.gitignore
```
```suggestion
```

pstarconsult replied:
Please proceed.
